diff --git a/Makefile b/Makefile
index a8b6835..f7bcaf4 100644
--- a/Makefile
+++ b/Makefile
@@ -291,7 +291,17 @@ else ifeq ($(platform), miyoo)
 	LIBM :=
 	FLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s
 	FLAGS += -DFAST_ALIGNED_LSB_WORD_ACCESS -fno-unroll-loops
-
+# TRIMUI MODEL S
+else ifeq ($(platform), trimui)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   fpic :=
+   SHARED := -shared -Wl,--version-script=link.T -Wl,--no-undefined
+   FLAGS += -fomit-frame-pointer -ffast-math
+   FLAGS += -mcpu=arm926ej-s -mtune=arm926ej-s
+   FLAGS += -DFAST_ALIGNED_LSB_WORD_ACCESS -fno-unroll-loops
 # (armv7 a7, hard point, neon based) ### 
 # NESC, SNESC, C64 mini 
 else ifeq ($(platform), classic_armv7_a7)
